c++/src/kj/compat/tls.c++

```diff
@@ -160,11 +161,21 @@ public:
       }
     });
   }
+
   kj::Promise<void> accept() {
     // We are the server. Set SSL options to prefer server's cipher choice.
     SSL_set_options(ssl, SSL_OP_CIPHER_SERVER_PREFERENCE);
 
-    return sslCall([this]() { return SSL_accept(ssl); }).ignoreResult();
+    return sslCall([this]() {
+             return SSL_accept(ssl);
+           })
+        .then([](size_t ret) {
+          if (ret == 0) {
+            kj::throwRecoverableException(
+                KJ_EXCEPTION(DISCONNECTED, "Client disconnected during SSL_accept()"));
+          }
+          KJ_REQUIRE(ret > 0, "sslCall must return a non-negative value");
```

Comment:
Yeah, I agree that it can't fail as the code is written now. I don't like an unhandled range though. I think this might make both of us feel better if I change the return type for sslCall.

Reply:
Waiiiiit, it's size_t anyway. This isn't just redundant, it's impossible.